Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {useContestCategories} from '../../../hooks/use-contest-categories';
+import Heading from '../../guidelines/headings/Heading';
+
+import styles from './ContestCategories.module.scss';
+import {FilterType} from '../../../common/contest-types';
+import {isEmpty} from 'lodash';
+import {useContests} from '../../../hooks/use-contests';
+import {generateFilterItems} from "../../../common/filter-utils";
+import {IHaveOptionalClassName} from "../../common/Props";
+import concatClassNames from "../../../utils/class-names";
+import Tree, {ITreeItemType} from "../../guidelines/trees/Tree";
+
+interface IContestCategoriesProps extends IHaveOptionalClassName {
+}
+
+const ContestCategories = ({
+    className = ''
+}: IContestCategoriesProps) => {
+    const {
+        state: {
+            categories,
+        },
+    } = useContestCategories();
+
+    const {
+        actions: {
+            applyFilter,
+        }
+    } = useContests();
+
+    const handleTreeItemClick = (node: ITreeItemType) => {
+        if (!isEmpty(node.children)) {
+            return;
+        }
+        
+        const filter = generateFilterItems(
+            FilterType.Category,
+            { name: node.name, value: node.id })[0];
```

Comment:
This is hacky - calling `[0]` at the end.
Also, `useContests` should have the filters with there generated `Id`s. I see no reason to generate the `selected` filter again.